chore(docs): Added step monitor doc (#1911)
newrelic/terraform-provider-newrelic: website/docs/r/synthetics_monitor_step.html.markdown

---
layout: "newrelic"
page_title: "New Relic: newrelic_synthetics_step_monitor"
sidebar_current: "docs-newrelic-resource-synthetics-step-monitor"
description: |-
Create and manage a Synthetics monitor in New Relic.
---

# Resource: newrelic\_synthetics\_step\_monitor

Use this resource to create, update, and delete the synthetics step monitor in New Relic.

> -> **NOTE:** The [newrelic_synthetics_private_location](newrelic_synthetics_private_location.html) resource private minion can take upto 10 minutes to be available through Terraform.

Reviewer: Take a look at the Additional Examples for the dashboard resource [https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/one_dashboard#additional-examples](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/one_dashboard#additional-examples)

Let's add something like that for a monitor with a private location and move this note there

> -> **NOTE:** The [newrelic_synthetics_private_location](newrelic_synthetics_private_location.html) resource private minion can take upto 10 minutes to be available through Terraform.

Reviewer: ```suggestion
-> **NOTE:** The [newrelic_synthetics_private_location](newrelic_synthetics_private_location.html) resource private minion can take up to 10 minutes to be available to use with a Synthetics monitor.
```

## Example Usage

##### Type: `STEP MONITOR`
```hcl
resource "newrelic_synthetics_step_monitor" "foo" {
  name = "foo"
  enable_screenshot_on_failure_and_script = true
  locations_public = ["US_EAST_1", "US_EAST_2"]
  period = "EVERY_6_HOURS"
  status = "ENABLED"
  steps {

>     ordinal = " "

Reviewer: Can we add more information to the steps? Ideally a few steps with different examples.

    types = " "
    values = "ASSERT_ELEMENT"
  }
  tag {
    key = "some_key"
    values = ["some_value"]
  }
}
```
See additional [examples](#additional-examples).

```hcl

> resource "newrelic_synthetics_private_location" "bar1" {

Reviewer: ```suggestion
resource "newrelic_synthetics_private_location" "private_location" {
```

  description               = "Test Description"

>   name                      = "%[1]A"

Reviewer: ```suggestion
  name                      = "private_location"
```

  verified_script_execution = true
}
  resource "newrelic_synthetics_step_monitor" "bar" {
    name             = "step_monitor"
    uri              = "https://www.one.example.com"

>     locations_public = ["AP_SOUTH_1"]

Reviewer: ```suggestion
      locations_private {
         guid         = newrelic_synthetics_private_location.private_location.id
         vse_password = secret
   }
```

    period           = "EVERY_6_HOURS"
    status           = "ENABLED"
    tag {
      key    = "some_key"
      values = ["some_value"]
    }
  }
```

> ## Argument Reference

Reviewer: Should also have `tag` and `steps` in this section with links to the nested blocks

Reviewer 2: Its already updated in the description in line 55 & 61

The following are the common arguments supported for `STEP` monitor:

> * `account_id`- (Required) The account in which the Synthetics monitor will be created.

Reviewer: ```suggestion
* `account_id`- (Optional) The account in which the Synthetics monitor will be created.
```

* `name` - (Required) The name for the monitor.
* `uri` - (Required) The uri the monitor runs against.

> * `locations_public` - (Required) The location the monitor will run from. Valid public locations are https://docs.newrelic.com/docs/synthetics/synthetic-monitoring/administration/synthetic-public-minion-ips/

Reviewer: Can we make a note that either only one of `locations_public` or `locations_private` is required?

* `locations_private` - (Required) The location the monitor will run from.
* `period` - (Required) The interval at which this monitor should run. Valid values are EVERY_MINUTE, EVERY_5_MINUTES, EVERY_10_MINUTES, EVERY_15_MINUTES, EVERY_30_MINUTES, EVERY_HOUR, EVERY_6_HOURS, EVERY_12_HOURS, or EVERY_DAY.
* `status` - (Required) The run state of the monitor. (i.e. `ENABLED`, `DISABLED`, `MUTED`).

> * `guid` - (Required) The unique identifier for the Synthetic Monitor in New Relic.

Reviewer: You can remove guid

### Nested blocks

All nested `locations_private` blocks support the following common arguments:

* `guid` - (Required) The unique identifier for the Synthetics private location in New Relic.

> * `vse_password` - (Optional) **DEPRECATED:** The location's Verified Script Execution password, Only necessary if Verified Script Execution is enabled for the location.

Reviewer: ```suggestion
* `vse_password` - (Optional) The location's Verified Script Execution password, Only necessary if Verified Script Execution is enabled for the location.
```

Reviewer 2: It will get deprecated for the new runtime

---
layout: "newrelic"
page_title: "New Relic: newrelic_synthetics_step_monitor"
sidebar_current: "docs-newrelic-resource-synthetics-step-monitor"
description: |-
Create and manage a Synthetics monitor in New Relic.
---

# Resource: newrelic\_synthetics\_step\_monitor

Use this resource to create, update, and delete the synthetics step monitor in New Relic.

## Example Usage

##### Type: `STEP MONITOR`
```hcl
resource "newrelic_synthetics_step_monitor" "foo" {
  name = "foo"
  enable_screenshot_on_failure_and_script = true

>   locations_public = ["AWS_US_EAST_1", "AWS_US_EAST_2"]

Author: remove AWS from the minion locations
